### PR TITLE
Updates repo topics URL for GHES docs

### LINKED ISSUE #34006: Repo topics URL points to GitHub.com in GHES docs
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/.github/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/enterprise-server@3.13/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics#about-topics

### What part(s) of the article would you like to see updated?

Specifically, there is a mention of finding top topics and it points to GitHub.com:

> To browse the most used topics, go to https://github.com/topics/.

However on GHES, this should be the hostname of the GHES instance.

image

### Additional information

This is on all GHES docs that I can find.

### PR DESCRIPTION
**Pull request by IAmHughes**
### Why:

GHES users shouldn't be pointed to the URL for GitHub.com topics. They should be pointed to their instance's topics list instead.

Closes: #34006 

### What's being changed (if available, include any code snippets, screenshots, or gifs):

I've updated the documentation to continue to link to the GitHub.com URL when on GHEC or FPT, but when on GHES it will display `https://HOSTNAME/topics` following the same nomenclature used in other documentation for GHES such as [about remote repositories](https://docs.github.com/en/enterprise-server@3.13/get-started/getting-started-with-git/about-remote-repositories#about-remote-repositories)

#### Old

image

#### New

image

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Review by Carlo0704 — COMMENTED** [review]
...

**Comment by nguyenalex836** [comment]
@IAmHughes Thanks so much for opening a PR! I'll get this triaged for review ✨

**Comment by MR-GIMA-MD2** [comment]
---
title: Configuring a publishing source for your GitHub Pages site
intro: 'You can configure your {% data variables.product.prodname_pages %} site to publish when changes are pushed to a specific branch, or you can write a {% data variables.product.prodname_actions %} workflow to publish your site.'
redirect_from:
  - /articles/configuring-a-publishing-source-for-github-pages
  - /articles/configuring-a-publishing-source-for-your-github-pages-site
  - /github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site
product: '{% data reusables.gated-features.pages %}'
permissions: 'People with admin or maintainer permissions for a repository can configure a publishing source for a {% data variables.product.prodname_pages %} site.'
versions:
  fpt: '*'
  ghes: '*'
  ghec: '*'
topics:
  - Pages
shortTitle: Configure publishing source
---

## About publishing sources

{% data reusables.pages.pages-about-publishing-source %}

{% data reusables.pages.private_pages_are_public_warning %}

## Publishing from a branch

1. Make sure the branch you want to use as your publishing source already exists in your repository.
{% data reusables.pages.navigate-site-repo %}
{% data reusables.repositories.sidebar-settings %}
{% data reusables.pages.sidebar-pages %}
1. Under "Build and deployment", under "Source", select **Deploy from a branch**.
1. Under "Build and deployment", use the branch dropdown menu and select a publishing source.
   Screenshot of Pages settings in a {% data variables.product.prodname_dotcom %} repository. A menu to select a branch for a publishing source, labeled "None," is outlined in dark orange.
1. Optionally, use the folder dropdown menu to select a folder for your publishing source.
   Screenshot of Pages settings in a {% data variables.product.prodname_dotcom %} repository. A menu to select a folder for a publishing source, labeled "/(root)," is outlined in dark orange.
1. Click **Save**.

### Troubleshooting publishing from a branch

{% data reusables.pages.symlink-removal %}

{% data reusables.pages.admin-must-push %}

If you choose the `docs` folder on any branch as your publishing source, then later remove the `/docs` folder from that branch in your repository, your site won't build and you'll get a page build error message for a missing `/docs` folder. For more information, see "[AUTOTITLE](/pages/setting-up-a-github-pages-site-with-jekyll/troubleshooting-jekyll-build-errors-for-github-pages-sites#missing-docs-folder)."

{% ifversion build-pages-with-actions %}

Your {% data variables.product.prodname_pages %} site will always be deployed with a {% data variables.product.prodname_actions %} workflow run, even if you've configured your {% data variables.product.prodname_pages %} site to be built using a different CI tool. Most external CI workflows "deploy" to {% data variables.product.prodname_pages %} by committing the build output to the `gh-pages` branch of the repository, and typically include a `.nojekyll` file. When this happens, the {% data variables.product.prodname_actions %} workflow will detect the state that the branch does not need a build step, and will execute only the steps necessary to deploy the site to {% data variables.product.prodname_pages %} servers.

To find potential errors with either the build or deployment, you can check the workflow run for your {% data variables.product.prodname_pages %} site by reviewing your repository's workflow runs. For more information, see "[AUTOTITLE](/actions/monitoring-and-troubleshooting-workflows/viewing-workflow-run-history)." For more information about how to re-run the workflow in case of an error, see "[AUTOTITLE](/actions/managing-workflow-runs/re-running-workflows-and-jobs)."

{% endif %}

## Publishing with a custom {% data variables.product.prodname_actions %} workflow

To configure your site to publish with {% data variables.product.prodname_actions %}:

{% data reusables.pages.navigate-site-repo %}
{% data reusables.repositories.sidebar-settings %}
{% data reusables.pages.sidebar-pages %}
1. Under "Build and deployment", under "Source", select **{% data variables.product.prodname_actions %}**.
1. {% data variables.product.product_name %} will suggest several starter workflows. If you already have a workflow to publish your site, you can skip this step. Otherwise, choose one of the options to create a {% data variables.product.prodname_actions %} workflow. For more information about creating your custom workflow, see "[Creating a custom {% data variables.product.prodname_actions %} workflow to publish your site](#creating-a-custom-github-actions-workflow-to-publish-your-site)."

   {% data variables.product.prodname_pages %} does not associate a specific workflow to the {% data variables.product.prodname_pages %} settings. However, the {% data variables.product.prodname_pages %} settings will link to the workflow run that most recently deployed your site.

### Creating a custom {% data variables.product.prodname_actions %} workflow to publish your site

For more information about {% data variables.product.prodname_actions %}, see "[AUTOTITLE](/actions)."

When you configure your site to publish with {% data variables.product.prodname_actions %}, {% data variables.product.product_name %} will suggest starter workflows for common publishing scenarios. The general flow of a workflow is to:

1. Trigger whenever there is a push to the default branch of the repository or whenever the workflow is run manually from the Actions tab.
1. Use the [`actions/checkout`](https://github.com/actions/checkout) action to check out the repository contents.
1. If required by your site, build any static site files.
1. Use the [`actions/upload-pages-artifact`](https://github.com/actions/upload-pages-artifact) action to upload the static files as an artifact.
1. If the workflow was triggered by a push to the default branch, use the [`actions/deploy-pages`](https://github.com/actions/deploy-pages) action to deploy the artifact. This step is skipped if the workflow was triggered by a pull request.

The starter workflows use a deployment environment called `github-pages`. If your repository does not already include an environment called `github-pages`, the environment will be created automatically. We recommend that you add a deployment protection rule so that only the default branch can deploy to this environment. For more information, see "[AUTOTITLE](/actions/deployment/targeting-different-environments/using-environments-for-deployment)."

{% ifversion fpt or ghec %}
{% note %}

**Note**: A `CNAME` file in your repository file does not automatically add or remove a custom domain. Instead, you must configure the custom domain through your repository settings or through the API. For more information, see "[AUTOTITLE](/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain)" and "[AUTOTITLE](/rest/pages#update-information-about-a-github-pages-site)."

{% endnote %}
{% endif %}

### Troubleshooting publishing with a custom {% data variables.product.prodname_actions %} workflow

For information about how to troubleshoot your {% data variables.product.prodname_actions %} workflow, see "[AUTOTITLE](/actions/monitoring-and-troubleshooting-workflows/about-monitoring-and-troubleshooting)."

**Comment by MR-GIMA-MD2** [comment]
{
"name": "mr gima",
"version": "2.1.2",
"description": "mr gima has been designed to make things easier and fun while using WhatsApp. 👨‍💻❤️",
"main": "start.js",
"scripts": {
"start": "pm2 start start.js --deep-monitoring --attach --name mr gima",
"stop": "pm2 stop mr gima",
"restart": "pm2 restart mr gima"
},
"dependencies": {
"@whiskeysockets/baileys": "git+https://github.com/PMD-API-UPDATE/Baileys-6.6.0.git",
"@adiwajshing/keyed-db": "^0.2.4",
"wa_set_pkg": "1.0.1",
"gemini-ai" : "latest",
"simple-git":"3.22.0",
"crypto-digest-sync": "^1.0.0",
"puppeteer": "^21.5.0",
"jspdf": "^2.5.1",
"imgur": "latest",
"mathjs": "^12.0.0",
"fraction.js": "^4.3.7",
"unzipper":"0.12.2",
"chart.js": "^4.4.0",
"qs": "^6.10.1",
"pino": "^7.0.5",
"google-tts-api": "^2.0.2",
"pm2": "^5.4.1",
"util": "^0.12.4",
"express": "latest",
"axios": "^1.6.7",
"file_size_url": "^1.0.4",
"cheerio": "^1.0.0-rc.12",
"@bochilteam/scraper": "4.2.4",
"fs-extra": "^11.1.0",
"api-dylux": "latest",
"mfiredlcore-vihangayt": "^1.0.0",
"path": "^0.12.7",
"form-data": "^4.0.0",
"@ffmpeg-installer/ffmpeg": "^1.1.0",  
"Sachibot-Spotifydl": "github:SACHIBOT/Sachibot-Spotifydl#master",
"ffmpeg": "^0.0.4",
"fluent-ffmpeg": "^2.1.2",
"crypto-js":"latest",
"youtubedl-core": "4.11.7",
"node-fetch": "^2.6.1" ,
"megajs": "1.1.0",
"file-type": "^16.5.3",
"vm": "^0.1.0",
"qrcode-terminal": "^0.12.0",
"sinhala-unicode": "2.0.0",
"wa-sticker-formatter": "^4.3.2",
"getscreenshot.js": "1.0.0",
"@blackamda/telegram-image-url": "^1.0.0",
"@sl-code-lords/esana-news": "1.0.3",
"@sl-code-lords/wabetainfo": "1.2.0",
"@sl-code-lords/photooxy": "1.0.1",
"google-image-sr": "^1.0.0",
"translatte": "^3.0.1",
"chatgpt-scraper": "^1.0.1",
"fouadwa-scraper": "2.0.0",
"xvideos-scraper": "1.0.4",
"@blackamda/song_video_dl": "1.1.0",
"ytdl-mp3": "3.4.0",
"lodash": "^4.17.21",
"google-it": "^1.6.4",
"tiktokdl":"0.0.4",
"@saipulanuar/scraper":"3.0.0",
"remove-bg-node":"1.0.3",
"qrcode":"1.5.3",
"qr-image":"3.2.0",
"jimp":"0.16.1",
"mathsteps":"^0.2.0",
"betabotz-tools":"0.0.6",
"tesseract.js":"2.0.1",
"@sl-code-lords/si-subdl":"1.1.1",
"@xct007/frieren-scraper":"github:xct007/frieren-scraper",
"ytdl-secktor":"latest",
"@sl-code-lords/movie-dl":"1.1.0",
"fb-info-video":"1.0.2",
"bard-ai":"2.0.3",
"scr-emoji":"0.0.3",
"emoji-api":"3.0.1",
"emoji-mixer":"1.0.33",
"@danitech/scraper":"1.0.7",
"dandi-api":"1.2.0",
"awesome-phonenumber":"6.2.0",
"vihangayt-fbdl":"1.0.2",
"@bochilteam/scraper-sosmed":"1.0.5",
"cc-suite":"1.0.5",
"btch-downloader":"2.3.1-alpha",
"buffer-url":"1.0.0",
"@sasmeee/wabetainfo":"1.1.1",
"nima-wabeta-info":"1.0.5",
"@vitalets/google-translate-api":"9.2.0",
"pastebin-js": "latest",
"is-standard-emoji":"1.0.0",
"@tobyg74/tiktok-api-dl":"1.0.13",
"sinhala-cricket-newsjs":"1.0.0",
"gms-mobile-news":"1.0.0",
"@octokit/rest":"20.0.2",
"mime-types":"2.1.35",
"@black-king/temp-mail-gen":"1.0.1",
"turndown":"7.1.3",
"imaginesdk":"latest",
"extract-zip":"2.0.1"
}
}

File: content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ With topics, you can explore repositories in a particular subject area, find pro
 
 ![Screenshot of the github/docs repository. In the right sidebar, three topics are outlined in dark orange: "docs," "hacktoberfest," and "works-with-codespaces."](/assets/images/help/repository/os-repo-with-topics.png)
 
-To browse the most used topics, go to https://github.com/topics/.
+To browse the most used topics, go to {% ifversion fpt or ghec %}https://github.com/topics/{% elsif ghec or ghes %}`https://HOSTNAME/topics/`{% endif %}.
```
Comment (lecoursen) [inline content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics.md:24]:
```suggestion
To browse the most used topics, go to {% data variables.product.product_url %}/topics/.
```

**Review by lecoursen — DISMISSED** [review]
Thanks @IAmHughes, very cool to see you again! 💖 I'm going to apply a small suggestion and then merge. ⚡

File: content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ With topics, you can explore repositories in a particular subject area, find pro
 
 ![Screenshot of the github/docs repository. In the right sidebar, three topics are outlined in dark orange: "docs," "hacktoberfest," and "works-with-codespaces."](/assets/images/help/repository/os-repo-with-topics.png)
 
-To browse the most used topics, go to https://github.com/topics/.
+To browse the most used topics, go to {% data variables.product.product_url %}/topics/.
```
Comment (lecoursen) [inline content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics.md:24]:
```suggestion
To browse the most used topics, go to https://{% data variables.product.product_url %}/topics/.
```

File: content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics.md (hunk 1 above)
Comment (IAmHughes) [reply content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics.md]: Hey @lecoursen , good to see you as well 👋 :octocat: 

It looks like this variable use may have broke the URLs on FPT/GHEC and then gives a bit of a weird format for GHES.

> GHEC - No hyperlink

image

> GHES - No code format and/or no `https://` (though I guess you can deploy GHES with `http`, to be fair)

image

---

I'll let you suggest another change if you prefer another format. For reference, I used the GHES template found in [about-remote-repositories](https://docs.github.com/en/enterprise-server@3.13/get-started/getting-started-with-git/about-remote-repositories#about-remote-repositories)

File: content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ With topics, you can explore repositories in a particular subject area, find pro
 
 ![Screenshot of the github/docs repository. In the right sidebar, three topics are outlined in dark orange: "docs," "hacktoberfest," and "works-with-codespaces."](/assets/images/help/repository/os-repo-with-topics.png)
 
-To browse the most used topics, go to https://github.com/topics/.
+To browse the most used topics, go to https://{% data variables.product.product_url %}/topics/.
```
Comment (lecoursen) [inline content/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics.md:24]:
```suggestion
To browse the most used topics, go to {% data variables.product.oauth_host_code %}/topics/.
```

**Review by lecoursen — APPROVED** [review]

**Pull request merged**